### PR TITLE
[bug 2152] [Security]: Filestore APIs should have authentication

### PR DESCRIPTION
**Pull request by ashishbeehyv**
https://github.com/pucardotorg/dristi/issues/2152
changes made: added auth token in headers of react-doc-viewer's DocViewer component

## Summary by CodeRabbit

- **New Features**
	- Enhanced document downloading functionality with improved error handling and security.
	- Document viewer now supports authenticated requests for secure document retrieval.
  
- **Bug Fixes**
	- Improved robustness of file download process by utilizing updated methods for handling requests.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+      const blob = new Blob([response.data], { type: mimeType });
+      const blobUrl = URL.createObjectURL(blob);
+
+      const link = document.createElement("a");
+      link.href = blobUrl;
+      link.download = `downloadedFile.${extension}`;
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+      URL.revokeObjectURL(blobUrl);
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js:32]: nice work :+1:

**Review by anirudh-0 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
-    link.click();
-    document.body.removeChild(link);
+
+    try {
+      const response = await axios.get(url, {
+        responseType: "blob",
+        headers: {
+          "auth-token": `${token}`,
+        },
+      });
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js:19]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+      const link = document.createElement("a");
+      link.href = blobUrl;
+      link.download = `downloadedFile.${extension}`;
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+      URL.revokeObjectURL(blobUrl);
+    } catch (error) {
+      console.error("Error downloading the PDF:", error);
+    }
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js:35]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
-
+    const token = localStorage.getItem("token");
     const url = `${window.location.origin}${Urls.FileFetchById}?tenantId=${tenantId}&fileStoreId=${fileStoreId}`;
-    const link = document.createElement("a");
-    link.href = url;
-    link.download = true;
-    document.body.appendChild(link);
-    link.click();
-    document.body.removeChild(link);
+
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useDownloadCasePdf.js:12]: not needed

**Pull request merged**